bokeh/util/structure.py

```diff
@@ -0,0 +1,315 @@
+"""Functions to create the graph of submodels of a model in networkx format,
+   and to draw that graph using bokeh so that one can explore the attributes of
+   submodels by clicking on the nodes.  Uses simple custom javascript callbacks so no server is necessary;
+   works in a jupyter notebook.
+
+   Suppose M is a bokeh model (i.e. a plot or figure).
+
+      - K = BokehStructureGraph(M) creates the graph, which is itself a bokeh model.
+      - K.show() calls show on the model to display it.
+      - K.model returns the model consisting of the structure graph and associated datatable.
+      - K.graph returns the directed graph in networkx format
+      - K.property_df returns the dataframe of models, submodels and their properties.
+"""
```

Comment:
Docstrings (here and throughout) should be updated to match existing project style:
```
""" Short description with a space in front of the sentence.

Further description. Not indented (aligned with docstring quotes).

Empty line at the end.

"""
```

Reply:
Done